frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -305,6 +315,140 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        /// Lock up and stake balance of the origin account.
+        ///
+        /// `value` must be more than the `minimum_balance` specified by `T::Currency`
+        /// unless account already has bonded value equal or more than 'minimum_balance'.
+        ///
+        /// The dispatch origin for this call must be _Signed_ by the staker's account.
+        ///
+        /// Effects of staking will be felt at the beginning of the next era.
+        ///
+        /// TODO: Weight!
+        #[pallet::weight(10)] // TODO: fix this later. Probably a new calculation will be required since logic was changed significantly.
+        pub fn bond_and_stake(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+            #[pallet::compact] value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let staker = ensure_signed(origin)?;
+            ensure!(
+                Self::is_contract_valid(&contract_id),
+                Error::<T>::ContractIsNotValid
+            );
+            ensure!(
+                RegisteredDapps::<T>::contains_key(&contract_id),
+                Error::<T>::NotOperatedContract
+            );
+
+            // Get the staking ledger or create an entry if it doesn't exist.
+            let mut ledger = if let Some(ledger) = Self::ledger(&staker) {
+                ledger
+            } else {
+                StakingLedger {
+                    stash: staker.clone(),
+                    total: Zero::zero(),
+                    active: Zero::zero(),
+                    unlocking: vec![],
+                    last_reward: Zero::zero(),
+                }
+            };
```

Comment:
@Maar-io - I think we should delete this storage. Since we don't have the bond/unbond/stake/withdraw granularity, this looses it's purpose. Plus, the agreement was that unbond_and_withdraw will be effective immediately.

We'll probably need it in the future but at this moment, it seems useless.

Reply:
We should still be able to easily add unbonding period for a few eras. We will start with 0 unbonding period but for the stability of the staked funds it might be decided to set unbonding period to few eras

Reply:
Ok, but keep in mind that this code doesn't make it much easier to add unbonding it in the future.

We will have to change the API and split this method into 2. That will be much more work than returning this ledger.

Reply:
ok, I think I misunderstood your question. I think we should consider removing ledger since we keep similar data in the ContractEraStake.stakers

Reply:
We can do it in some follow-up PR so we don't delay this one.